packages/handleRichTextBoxKeyEventNavigation/handleRichTextBoxKeyEventNavigation.js

```diff
@@ -44,3 +72,42 @@ export default function(e, record) {
     }
     return false;
 }
+
+// Depends on the implementation, there might be nested elements for a record's
+// rich text box DOMNode. Therefore, we need to skip all elements contained in
+// the current record's DOMNode to navigate to a focusable element outside the
+// record
+function findAdjcentFocusableElem_(record, findPrev) {
+    const tabbableElems = [
+        ...document.querySelectorAll(TABBABLE_ELEMENT_SELECTOR),
+    ];
+    const recordDom = record.getDom();
+    if (!recordDom) {
+        return null;
+    }
+
+    const recordElems = [
+        ...recordDom.querySelectorAll(TABBABLE_ELEMENT_SELECTOR),
+    ];
+
+    const tabbableEls = findPrev ? tabbableElems : tabbableElems.reverse();
+
+    for (let i = 0; i < tabbableEls.length; i++) {
```

Comment:
This is readable as is so I wouldn't suggest changing it - however this will be slower if findPrev is false since reverse is O(N). If you wanted to be clever, you could instead change the loop:

```
if (findPrev) {
    for (let i = 0; i < tabbableElems.length; i++) {
        processElement(tabbableElems[i], i)
    }
} else {
    for (let i = tabbableElems.length - 1; i >= 0; i--) {
        processElement(tabbableElems[i], i)
    }
}
```

Or you could just invert the index

```
for (let n = 0; n < tabbableElems.length; n++) {
	const idx = findPrev ? n : (tabbableElems.length - 1) - n
	// existing code
}
```

Reply:
hmmm I had A hard time making a trade-off between perf and readability. But I don't like `reserve()` either. Changed it.